client/repo.c

```diff
@@ -555,6 +555,48 @@ TDNFGetRepoMD(
                   pszRepoDataDir);
     BAIL_ON_TDNF_ERROR(dwError);
 
+    if (pRepoData->pszMirrorList) {
+        time_t now = time(NULL);
+        int needDownload = 0;
+        struct stat st = {0};
+
+        dwError = TDNFGetCachePath(pTdnf, pRepoData,
+                                   TDNF_REPO_METADATA_MIRRORLIST, NULL,
+                                   &pszMirrorFile);
+        BAIL_ON_TDNF_ERROR(dwError);
+
+        if (stat(pszMirrorFile, &st) < 0) {
+            if (errno == ENOENT)
+                needDownload = 1;
+            else {
+                dwError = errno;
+                BAIL_ON_TDNF_SYSTEM_ERROR(dwError);
+            }
+        } else if ((now - st.st_ctime) > pRepoData->lMetadataExpire)
```

Comment:
what happens if system time is wrong?

Reply:
Nothing we can do about that, we don't have any superpowers. This is the same as for repo data expiry.